src/main/java/org/gridsuite/loadflow/server/service/LoadFlowWorkerService.java

```diff
@@ -108,21 +68,23 @@ private CompletableFuture<LoadFlowResult> runLoadFlowAsync(
             CompletableFuture<LoadFlowResult> future = runner.runAsync(
                     network,
                     variantId != null ? variantId : VariantManagerConstants.INITIAL_VARIANT_ID,
-                    loadFlowExecutionService.getComputationManager(),
+                    executionService.getComputationManager(),
                     params,
                     reporter);
             if (resultUuid != null) {
                 futures.put(resultUuid, future);
             }
             return future;
         } finally {
-            lockRunAndCancelLF.unlock();
+            lockRunAndCancel.unlock();
         }
     }
 
-    private LoadFlowResult run(Network network, LoadFlowRunContext context, UUID resultUuid) throws Exception {
-        LoadFlowParameters params = buildParameters(context.getParameters(), context.getProvider());
+    @Override
+    protected LoadFlowResult run(LoadFlowRunContext context, UUID resultUuid) throws Exception {
```

Comment:
I think it'possible to have a generic method

Reply:
I tried to and it complicated a lot without encouraging results;

Examples : 

```
LoadFlowResult result = future == null ? null : observer.observeRun("run", context, future::get);
        if (result != null && result.isOk()) {
```
LoadFlowResult is a powsybl types. How do I handle the call to "isOk" ?

Same kind of problem with this inside run : 
`LoadFlowParameters params = buildParameters(context);`

Moreover the run code is sometimes very different in the other computation microservices; Example in security analysis, there is the handling of contingencies : 
 ```
contingencies.stream()
                    .filter(contingencyInfos -> !CollectionUtils.isEmpty(contingencyInfos.getNotFoundElements()))
                    .forEach(contingencyInfos -> {
                        String elementsIds = String.join(", ", contingencyInfos.getNotFoundElements());
                        notFoundElementReports.add(Report.builder()
                                .withKey("contingencyElementNotFound_" + contingencyInfos.getId() + notFoundElementReports.size())
                                .withDefaultMessage(String.format("Cannot find the following equipments %s in contingency %s", elementsIds, contingencyInfos.getId()))
                                .withSeverity(TypedValue.WARN_SEVERITY)
                                .build());
                    });
            if (!CollectionUtils.isEmpty(notFoundElementReports)) {
                Reporter elementNotFoundSubReporter = reporter.createSubReporter(context.getReportUuid().toString() + "notFoundElements", "Elements not found");
                notFoundElementReports.forEach(elementNotFoundSubReporter::report);
            }
```